samila/genimage.py

```diff
@@ -109,12 +109,18 @@ def generate(
                 elif self.generate_mode == "f2_vs_f1":
                     data2_ = self.function1(point[0], point[1]).real
                     data1_ = self.function2(point[0], point[1]).real
-                elif self.generate_mode == "f2":
+                elif self.generate_mode == "f2_vs_index":
                     data2_ = index
                     data1_ = self.function2(point[0], point[1]).real
-                elif self.generate_mode == "f1":
+                elif self.generate_mode == "f1_vs_index":
                     data2_ = index
                     data1_ = self.function1(point[0], point[1]).real
+                elif self.generate_mode == "index_vs_f1":
+                    data1_ = index
+                    data2_ = self.function1(point[0], point[1]).real
+                elif self.generate_mode == "index_vs_f2":
+                    data1_ = index
+                    data2_ = self.function2(point[0], point[1]).real
```

Comment:
Isn't it better to compare them with `GenerateMode.X.value`?

Reply:
fixed in https://github.com/sepandhaghighi/samila/pull/222/commits/e5a0f255240496c83f1cf7ea70395d86e93e5401